Vulnerability Scanning Implementation for container images (#1489)
shipwright-io/build: docs/build.md

    - Use string `SourceTimestamp` to set the image timestamp to the source timestamp, i.e. the timestamp of the Git commit that was used.
    - Use string `BuildTimestamp` to set the image timestamp to the timestamp of the build run.
    - Use any valid UNIX epoch seconds number as a string to set this as the image timestamp.

>   - `spec.output.vulnerabilityScan` to provides configurations about running a scan for your generated image. Furthur options in vulnerability scanning are defined [here](#defining-the-vulnerabilityscan)

Reviewer: I just checked, "Furthur" is an American rock band. I did not know them. ;-)

```suggestion
  - `spec.output.vulnerabilityScan` to provides configurations about running a scan for your generated image. Further options in vulnerability scanning are defined [here](#defining-the-vulnerabilityscan)
```

>   - `spec.output.vulnerabilityScan` to provides configurations about running a scan for your generated image. Further options in vulnerability scanning are defined [here](#defining-the-vulnerabilityscan)

Reviewer: ```suggestion
  - `spec.output.vulnerabilityScan` to enable a security vulnerability scan for your generated image. Further options in vulnerability scanning are defined [here](#defining-the-vulnerabilityscan)
```

    timestamp: SourceTimestamp
```

### Defining the vulnerabilityScan

> `vulnerabilityScan` provides configurations about running a scan for your generated image.

Reviewer: ```suggestion
`vulnerabilityScan` provides configurations to run a scan for your generated image.
```

> - `vulnerabilityScan.enabled` - Specify whether to run vulnerability scan for image. The supported types are true and false.

Reviewer: ```suggestion
- `vulnerabilityScan.enabled` - Specify whether to run vulnerability scan for image. The supported values are true and false.
```

> - `vulnerabilityScan.fail` - indicates whether to fail the build run if the vulnerability scan results in vulnerabilities. The supported types are true and false.

Reviewer: ```suggestion
- `vulnerabilityScan.fail` - indicates whether to fail the build run if the vulnerability scan results in vulnerabilities. The supported values are true and false. This field is optional and false by default.
```

- `vulnerabilityScan.ignore.issues` - references the security issues to be ignored in vulnerability scan

> - `vulnerabilityScan.ignore.severity` - indicates the severities of security issues to be ignored (comma separated). Supported types are : Low, Medium, High, Critical.

Reviewer: This is not what I intended. In the ship, I wrote:

```yaml
  severity: Low | Medium | High | Critical
```

So, the value is either low or medium or high or critical, but not multiple of them. Then, we would chose an array. Severities have an order. If Low is specified, then only Low should be ignored. If Medium is defined, then Low and Medium are ignored. Etc.

That was my idea.

Side note: while just checking this in the SHIP, I noticed that I used Low | Medium | High | Critical in the spec, while I used lower case medium and high in the status. We should be consistent and I tend to suggest to use fully lower-case everywhere.

Author: @SaschaSchwarze0 Shall we allow Critical here ? because if someone specifies critical, then it will ignore all the vulnerabilities.
